src/util/taproot.rs

```diff
@@ -543,9 +543,9 @@ impl NodeInfo {
 
     /// Creates a new leaf [`NodeInfo`] with given [`Script`] and [`LeafVersion`].
     pub fn new_leaf_with_ver(script: Script, ver: LeafVersion) -> Self {
-        let leaf = LeafInfo::new(script, ver);
+        let leaf = ScriptLeaf::new(script, ver);
         Self {
-            hash: leaf.hash(),
+            hash: sha256::Hash::from_inner(leaf.leaf_hash().into_inner()),
```

Comment:
Perhaps for RC fix consider `impl From<TapLeafhash> for Hash`

Reply:
This is not a topic of unrelated separate discussions. All hashes in taptree are tagged. I think we will need a newtype over sha256 hash `TapNodeHash`, which will mean "any type of taptree node hash", but not other sha256 hash - and then have `From` TapLeaf/BranchHash to `TapNodeHash`. But this is for the next release probably.

Reply:
That makes sense.